pkgs/io_file/test/read_as_bytes_test.dart

```diff
@@ -0,0 +1,229 @@
+
+      final data = randomUint8List(20);
+      File(path1).writeAsBytesSync(data);
+      Link(path2).createSync(path1);
+
+      expect(fileSystem.readAsBytes(path2), data);
+    });
+
+    test('broken symlink', () {
+      final path1 = '$tmp/file1';
+      final path2 = '$tmp/file2';
+
+      final data = randomUint8List(20);
+      File(path1).writeAsBytesSync(data);
+      Link(path2).createSync(path1);
+      File(path1).deleteSync();
+
+      expect(
+        () => fileSystem.readAsBytes(path2),
+        throwsA(
+          isA<PathNotFoundException>()
+              .having((e) => e.message, 'message', 'open failed')
+              .having(
+                (e) => e.osError?.errorCode,
+                'errorCode',
+                2, // ENOENT
+              )
+              .having((e) => e.path, 'path', path2),
+        ),
+      );
+    });
+
+    group('fifo (does not have length)', () {
+      for (var i = 0; i <= 1024; ++i) {
+        test('Read small file: $i bytes', () async {
+          final data = randomUint8List(i);
+          final path = '$tmp/file';
+          stdlibc.mkfifo(path, 438); // 0666
+
+          (await FifoWriter.create(path))
+            ..write(data)
+            ..close();
+
+          expect(fileSystem.readAsBytes(path), data);
+        });
+      }
+
+      test('many single byte reads', () async {
+        final path = '$tmp/file';
+        stdlibc.mkfifo(path, 438); // 0666
+
+        final writer = await FifoWriter.create(path);
+        final data = randomUint8List(20);
+
+        for (var byte in data) {
+          writer
+            ..write(Uint8List(1)..[0] = byte)
+            ..delay(const Duration(milliseconds: 10));
+        }
+        writer.close();
+        expect(fileSystem.readAsBytes(path), data);
+      });
+
+      for (var i = blockSize - 2; i <= blockSize + 2; ++i) {
+        test('Read close to `blockSize`: $i bytes', () async {
+          final data = randomUint8List(i);
+          final path = '$tmp/file1';
+          stdlibc.mkfifo(path, 438); // 0666
+
+          (await FifoWriter.create(path))
+            ..write(data)
+            ..close();
+          expect(fileSystem.readAsBytes(path), data);
+        });
+      }
+    });
+
+    group('regular files', () {
+      for (var i = 0; i <= 1024; ++i) {
+        test('Read small file: $i bytes', () {
+          final data = randomUint8List(i);
+          final path = '$tmp/file';
+
+          File(path).writeAsBytesSync(data);
+          expect(fileSystem.readAsBytes(path), data);
+        });
+      }
+
+      for (var i = maxReadSize - 2; i <= maxReadSize + 2; ++i) {
+        test('Read close to `maxReadSize`: $i bytes', () {
+          final data = randomUint8List(i);
+          final path = '$tmp/file1';
+
+          File(path).writeAsBytesSync(data);
+          expect(fileSystem.readAsBytes(path), data);
+        });
+      }
+
+      test('very large file', () {
+        final data = randomUint8List(1 << 31 + 1);
```

Comment:
Maybe use `2 * 1024 * 1024 * 1024 + 1` for readability.

Reply:
Added a comment clarifying that I'm picking a number that is bigger than `INT_MAX`.